fix RP definition in the HW watchdog use case (#243)
ietf-rats-wg/architecture: draft-ietf-rats-architecture.md

Relying Party:

> : The watchdog capable to trigger a procedure that resets a device into
> a known, good operational state.

Reviewer: ```suggestion
: A watchdog capable of triggering a procedure that resets a device into
a known, good operational state.
```

Reviewer 2: +1
